util/compression_test.cc

```diff
@@ -220,6 +220,141 @@ TEST_F(DBAutoSkip, AutoSkipCompressionManager) {
     ASSERT_OK(Flush());
   }
 }
+class CostAwareTestFlushBlockPolicy : public FlushBlockPolicy {
+ public:
+  explicit CostAwareTestFlushBlockPolicy(const int window,
+                                         const BlockBuilder& data_block_builder)
+      : window_(window),
+        num_keys_(0),
+        data_block_builder_(data_block_builder) {}
+
+  bool Update(const Slice& /*key*/, const Slice& /*value*/) override {
+    auto nth_window = num_keys_ / window_;
+    if (data_block_builder_.empty()) {
+      // First key in this block
+      return false;
+    }
+    // Check every window
+    if (num_keys_ % window_ == 0) {
+      auto set_compress_time_and_size = [](size_t time, size_t size) {
+        return [time, size](void* arg) {
+          std::pair<size_t, size_t>* measured_data =
+              static_cast<std::pair<size_t, size_t>*>(arg);
+          measured_data->first = time;
+          measured_data->second = size;
+        };
+      };
+      auto get_predictor = [&](void* arg) {
+        auto predictor = static_cast<IOCPUCostPredictor*>(arg);
+        predicted_cpu_time_ = predictor->CPUPredictor.Predict();
+        predicted_io_bytes_ = predictor->IOPredictor.Predict();
+      };
+      SyncPoint::GetInstance()->DisableProcessing();
+      SyncPoint::GetInstance()->ClearAllCallBacks();
+
+      // Add syncpoint to get the cpu and io cost
+      SyncPoint::GetInstance()->SetCallBack(
+          "CostAwareCompressor::CompressBlockAndRecord::"
+          "GetPredictor",
+          get_predictor);
+      // use nth window to detect test cases and set the expected
+      switch (nth_window) {
+        case 0:
+          SyncPoint::GetInstance()->SetCallBack(
+              "CostAwareCompressor::CompressBlockAndRecord::"
+              "SetCompressionTimeOutputSize",
+              set_compress_time_and_size(1000, 100));
```

Comment:
This is also an opportunity to use some test function at least on predictor->TEST_ MockCompressTimeAndSize(1000, 100) instead of using a callback, since you already have predictor to you through sync point above.